ports/samd/samd_flash.c

```diff
@@ -82,9 +76,17 @@ STATIC mp_obj_t samd_flash_init(void) {
     #endif
 
     flash_init(&flash_desc, NVMCTRL);
-    return mp_const_none;
 }
-STATIC MP_DEFINE_CONST_FUN_OBJ_0(samd_flash_init_obj, samd_flash_init);
+
+// FLASH stuff
+STATIC mp_obj_t samd_flash_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    // No args required. bdev=Flash(). Start Addr & Size defined in samd_flash_obj.
+    mp_arg_check_num(n_args, n_kw, 0, 0, false);
+
+    samd_flash_init();
```

Comment:
Is it OK if this init is called more than once?  Eg on soft-reset the `_boot.py` will run again and re construct the flash object, calling this function again.

I guess it must be OK because prior to this PR the `flash.init()` method was called each time `_boot.py` ran.

Reply:
That#s uncritical. samd_flash_init() does nothing but enabling the clock for the device by setting a bit in a register. And that can be done multiple times.